### ISSUE TITLE
[Accessibility][Meta] Headings in examples and on pages should be accessible

### ISSUE THREAD

**Issue by citelao** [microsoft/WinUI-Gallery#1224]
### Which version of the app?

WinUI 3 Gallery

### Description

There are visual headings all over the gallery, but they are typically not accessible as headings.

We should:

* Add `AutomationProperties.HeadingLevel` to each heading.
* Make a Style to allow folks to easily add accessible headings in code.
